x/gamm/pool-models/stableswap/amm.go

```diff
@@ -276,6 +281,59 @@ func solveCfmmMulti(xReserve, yReserve, wSumSquares, yIn osmomath.BigDec) osmoma
 	return a
 }
 
+// solidly CFMM is xy(x^2 + y^2) = k
+// So we want to solve for a given addition of `b` units of y into the pool,
+// how many units `a` of x do we get out.
+// Let y' = y + b
+// we solve k = (x'y')(x'^2 + y^2) for x', using the following equation {wolfram alpha link}
```

Comment:
Needs wolfram alpha link, or equation we test and explanation of correctness

Reply:
I put a wolfram alpha link of the original equation, a desmos link for our simplified one, and direct equation + our key abstractions in my last commit